src/features/website/header/view/HeaderMobile.tsx

```diff
@@ -0,0 +1,62 @@
+'use client';
+
+import { useCallback, useEffect, useState } from 'react';
+
+import type {
+    MantineBreakpoint,
+    Box,
+    Group,
+    Burger,
+    type BoxProps,
+} from '@mantine/core';
+import { useMediaQuery } from '@mantine/hooks';
+
+import { Actions } from '@/features/website/header/view/Actions';
+import { DrawerMenu } from '@/features/website/header/view/DrawerMenu';
+import { BREAKPOINTS } from '@/shared/config/breakpoints';
+import { Brand } from '@/shared/ui/Brand';
+
+interface HeaderMobileProps extends BoxProps {
+    hiddenFrom: MantineBreakpoint;
+}
+
+export const HeaderMobile = ({ hiddenFrom }: HeaderMobileProps) => {
+    const [open, setOpen] = useState(false);
+    const handleBurgerClick = useCallback(() => setOpen((v) => !v), []);
```

Comment:
```suggestion
    const handleBurgerClick = useCallback(() => setOpen((isOpen) => !isOpen), []);
```

Додам, що краще використовувати більш зрозумілий неймінг для змінних, а взагалі про колбеки та useDisclosure вірно